ownership-chain/runtime/laos/src/configs/fee.rs

```diff
@@ -0,0 +1,215 @@
+		type FreezeIdentifier = ();
+		type MaxHolds = ();
+		type MaxFreezes = ();
+	}
+	impl pallet_authorship::Config for Test {
+		type FindAuthor = AuthorGiven;
+		type EventHandler = ();
+	}
+
+	const TEST_ID: ConsensusEngineId = [1, 2, 3, 4];
+
+	pub struct AuthorGiven;
+
+	impl frame_support::traits::FindAuthor<u64> for AuthorGiven {
+		fn find_author<'a, I>(digests: I) -> Option<u64>
+		where
+			I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>,
+		{
+			for (id, mut data) in digests {
+				if id == TEST_ID {
+					return u64::decode(&mut data).ok()
+				}
+			}
+
+			None
+		}
+	}
+
+	fn seal_header(mut header: Header, author: u64) -> Header {
+		{
+			let digest = header.digest_mut();
+			digest.logs.push(DigestItem::PreRuntime(TEST_ID, author.encode()));
+			digest.logs.push(DigestItem::Seal(TEST_ID, author.encode()));
+		}
+
+		header
+	}
+
+	fn create_header(number: u64, parent_hash: H256, state_root: H256) -> Header {
+		Header::new(number, Default::default(), state_root, parent_hash, Default::default())
+	}
+
+	fn new_test_ext() -> sp_io::TestExternalities {
+		let t = frame_system::GenesisConfig::<Test>::default().build_storage().unwrap();
+		t.into()
+	}
+
+	fn initialize_block_and_set_author(number: u64, author: AccountId) {
+		let mut header =
+			seal_header(create_header(number, Default::default(), [1; 32].into()), author);
+
+		header.digest_mut().pop(); // pop the seal off.
+		System::reset_events();
+		System::initialize(&number, &Default::default(), header.digest());
+	}
+
+	#[test]
+	fn sets_author_lazily() {
+		new_test_ext().execute_with(|| {
+			let author = 42;
+
+			initialize_block_and_set_author(1, author);
+			assert_eq!(Authorship::author(), Some(author));
+		});
+	}
+
+	#[test]
+	fn all_fee_should_go_to_block_author() {
+		new_test_ext().execute_with(|| {
+			let fee_amount = 100;
+			let author = 62;
+
+			initialize_block_and_set_author(1, author);
+
+			// Check initial conditions
+			let initial_author_balance = pallet_balances::Pallet::<Test>::free_balance(author);
+			assert_eq!(initial_author_balance, 0, "Initial author balance should be 0");
+
+			// Mock the creation of a negative imbalance and distribute fees
+			let imbalance = pallet_balances::Pallet::<Test>::issue(fee_amount);
+			ToAuthor::<Test>::on_unbalanceds(vec![imbalance].into_iter());
+
+			// Assert the final state
+			let final_author_balance = pallet_balances::Pallet::<Test>::free_balance(author);
+			assert_eq!(
+				final_author_balance, fee_amount,
+				"Author did not receive the correct amount"
+			);
+		});
+	}
+
+	#[test]
+	fn issuance_should_not_change_after_fee_distribution() {
+		new_test_ext().execute_with(|| {
+			let fee_amount = 100;
+			let author = 62;
+
+			initialize_block_and_set_author(1, author);
+
+			// Mock the creation of a negative imbalance
+			let imbalance = pallet_balances::Pallet::<Test>::issue(fee_amount);
+
+			let initial_total_issuance = pallet_balances::Pallet::<Test>::total_issuance();
```

Comment:
you have to get initial total issuance this before you call `issue()`, because `issue` increases the issuance. as suggested before, try to use `withdraw` if you want to have unchanged issuance

Reply:
https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/balances/src/impl_currency.rs#L276

Reply:
To execute a `withdraw`, a `deposit` is necessary first, followed by determining the initial total reference. 
Therefore, using the `issue()` function is deemed a logical approach.